File: mpp/LabMPPCurse/Models/src/main/java/Models/Cursa/ValidatorCursa.java
package Models.Cursa;

import Models.ValidationException;
import Models.Validator;

public class ValidatorCursa implements Validator<Cursa> {
    @Override
    public void validate(Cursa entity) throws ValidationException {

        String msg = "";
        if (entity.getNrLocuriDisponibile()< 0 || entity.getNrLocuriDisponibile() > 18)
        {
            msg += "numar locuri invalid";
        }

        if (entity.getIdDestinatie() <= 0)
        {
            msg += "destinatie invalida";
        }

        if (msg.compareTo("") != 0)
        {
            throw new ValidationException(msg);
        }
    }
}

File: mpp/CurseMPP/CurseMPP/Models/Cursa/ValidatorCursa.cs
namespace CurseMPP.Models
{
	public class ValidatorCursa : IValidator<Cursa>
	{
		public void validate(Cursa elem)
		{
			string msg = "";
			if (elem.nrLocuriDisponibile < 0 || elem.nrLocuriDisponibile > 18)
			{
				msg += "numar locuri invalid";
			}

			if (elem.idDestinatie <= 0)
			{
				msg += "destinatie invalida";
			}

			if (msg.CompareTo("") != 0)
			{
				throw new ValidationException(msg);
			}
		}
	}
}